src/snowflake/snowpark/_internal/data_source_utils.py

```diff
@@ -0,0 +1,347 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+
+import datetime
+import logging
+from enum import Enum
+from typing import List, Any, Tuple, Protocol
+from snowflake.connector.options import pandas as pd
+from dateutil import parser
+
+from snowflake.snowpark.exceptions import SnowparkDataframeReaderException
+from snowflake.snowpark.types import (
+    StringType,
+    GeographyType,
+    VariantType,
+    BinaryType,
+    GeometryType,
+    TimestampType,
+    DecimalType,
+    FloatType,
+    ShortType,
+    IntegerType,
+    BooleanType,
+    LongType,
+    TimeType,
+    ByteType,
+    DateType,
+    TimestampTimeZone,
+    StructType,
+    StructField,
+)
+
+_logger = logging.getLogger("snowflake.snowpark")
```

Comment:
usually we do __name__ to use the module name

```suggestion
_logger = getLogger(__name__)
```